compiler/src/iree/compiler/Codegen/Common/test/link_tuning_specs.mlir

```diff
@@ -135,3 +135,161 @@ module @td_module_4 attributes { transform.with_named_sequence } {
+        transform.match.operation_name %matmul ["linalg.generic"] : !transform.any_op
+
+        %lhs = transform.get_operand %matmul[0] : (!transform.any_op) -> !transform.any_value
+        %rhs = transform.get_operand %matmul[1] : (!transform.any_op) -> !transform.any_value
+        transform.iree.match.cast_compatible_type %lhs = tensor<?x?xf16> : !transform.any_value
+        transform.iree.match.cast_compatible_type %rhs = tensor<?x?xf16> : !transform.any_value
+
+        // Generate `config` inside this sequence
+        %config = transform.param.constant #iree_codegen.compilation_info<
+            lowering_config = #iree_gpu.lowering_config<{x}>,
+            translation_info = #iree_codegen.translation_info<pipeline = None>
+        > -> !transform.any_param
+
+        transform.yield %matmul, %config : !transform.any_op, !transform.any_param
+        }
+
+        transform.named_sequence
+        @__kernel_config(%variant_op: !transform.any_op {transform.consumed}) -> !transform.any_op
+        attributes { iree_codegen.tuning_spec_entrypoint } {
+        %res = transform.foreach_match in %variant_op
+            @match_mmt_f16_f16_f32 -> @apply_op_config
+            : (!transform.any_op) -> !transform.any_op
+        transform.yield %res : !transform.any_op
+        }
+    }
+
+    module @attention_module attributes { transform.with_named_sequence, iree_codegen.tuning_spec_with_default_entrypoint } {
+        transform.named_sequence @apply_attn_op_config(%attention: !transform.any_op {transform.readonly},
+                                                 %config: !transform.any_param {transform.readonly},
+                                                 %decomposition_config: !transform.any_param {transform.readonly}) {
+            transform.annotate %attention "compilation_info" = %config : !transform.any_op, !transform.any_param
+            transform.annotate %attention "decomposition_config" = %decomposition_config : !transform.any_op, !transform.any_param
+            transform.yield
+        }
+
+        transform.named_sequence @match_attention_f16(%attention: !transform.any_op {transform.readonly}) -> (!transform.any_op, !transform.any_param, !transform.any_param) {
+            transform.match.operation_name %attention ["iree_linalg_ext.attention"] : !transform.any_op
+            %in0 = transform.get_operand %attention[0] : (!transform.any_op) -> !transform.any_value
+            transform.iree.match.cast_compatible_type %in0 = tensor<?x?x?x?xf16> : !transform.any_value
+
+            %config = transform.param.constant #iree_codegen.compilation_info<
+                    lowering_config = #iree_gpu.lowering_config<{x}>,
+                    translation_info = #iree_codegen.translation_info<pipeline = None>>
+            -> !transform.any_param
+
+            %decomposition_config = transform.param.constant {
+            qk_attrs = {x},
+            pv_attrs = {y}
+            } -> !transform.any_param
```

Comment:
This test shouldn't have to do anything related to attention. Let's try to keep the linking tests minimal.